### ISSUE TITLE
docker datadog agent fails to retail logs on restarted container

### ISSUE THREAD

**Issue by dirkraft** [DataDog/datadog-agent#4364]
**Output of the info page (if this is a bug)**
```
root@ip-10-12-89-151:~# docker exec -it dd-agent s6-svstat /var/run/s6/services/agent/
up (pid 335) 61923 seconds
```

Nothing interesting there. Agent logs seem pretty clear.

```
2019-10-22 16:46:17 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:52 in Schedule) | Received a new logs config: docker
2019-10-22 16:46:17 UTC | CORE | WARN | (pkg/logs/input/docker/launcher.go:198 in startTailer) | Can't tail twice the same container: 01458476e561
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:84 in Unschedule) | New source to remove: entity: docker://01458476e561fa9aff5e81768a669c245aeb737fc9c9b0ea4c2c7e111bf71906
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:84 in Unschedule) | New source to remove: entity: docker://01458476e561fa9aff5e81768a669c245aeb737fc9c9b0ea4c2c7e111bf71906
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:99 in Unschedule) | New service to remove: entity: docker://01458476e561fa9aff5e81768a669c245aeb737fc9c9b0ea4c2c7e111bf71906
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/input/docker/tailer.go:80 in Stop) | Stop tailing container: 01458476e561
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/autodiscovery/config_poller.go:95 in poll) | docker provider: collected 0 new configurations, removed 1
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:84 in Unschedule) | New source to remove: entity: docker://01458476e561fa9aff5e81768a669c245aeb737fc9c9b0ea4c2c7e111bf71906
2019-10-22 16:46:21 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:84 in Unschedule) | New source to remove: entity: docker://01458476e561fa9aff5e81768a669c245aeb737fc9c9b0ea4c2c7e111bf71906
```

**Describe what happened:**
Monitored application in docker container crashed, container restarted because `restart: always` in compose config (shown below). dd-agent is still running and continues to accept metrics posted by the application, but no longer tailing docker logs. 

I think the datadog agent logs above are pretty clear: agent tries to re-tail logs before cleaning up its previous tail, so a check prevents it from doing that. Now our application has restarted itself but we are not seeing logs on datadog.com. Metrics are fine.

**Describe what you expected:**
datadog agent should re-tail logs on crashed & restarted docker container.

**Steps to reproduce the issue:**
Our compose file is essentially this, started via `docker-compose --file compose.yaml up -d`

```
version: '2'
services:
  author-disambiguation:
    restart: always
    image: "my-application"
    labels:
      # multiline expression based on
      # https://docs.datadoghq.com/logs/log_collection/java/?tab=log4j#configure-the-datadog-agent
      com.datadoghq.ad.logs: |
        [{
          "source": "docker",
          "service": "my-service",
          "log_processing_rules": [{
            "type": "multi_line",
            "name": "new_log_start_with_date",
            "pattern": "\\d{4}-\\d{1,2}-\\d{1,2}"
          }]
        }]
    depends_on:
      - dd-agent
  dd-agent:
    restart: always
    container_name: 'dd-agent'
    image: 'datadog/agent:latest-jmx'
    ports: ['8125:8125/udp']
    environment:
      - DD_API_KEY=${DD_API_KEY}
      - DD_USE_DOGSTATSD=true
      - DD_DOGSTATSD_PORT=8125
      - DD_DOGSTATSD_NON_LOCAL_TRAFFIC=true
      - DD_LOGS_ENABLED=true
      - DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true
      - DD_AC_EXCLUDE=name:dd-agent
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /proc/:/host/proc/:ro
      - /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
      - /root/conf.d/jmx.d:/etc/datadog-agent/conf.d/jmx.d:ro
```

**Additional environment details (Operating System, Cloud provider, etc):**

Here's some versions. Not sure it matters.
```
root@ip-10-12-89-151:~# docker --version
Docker version 18.06.1-ce, build e68fc7a

root@ip-10-12-89-151:~# docker-compose --version
docker-compose version 1.22.0, build f46880fe

root@ip-10-12-89-151:~# uname -a
Linux ip-10-12-89-151 4.15.0-1023-aws #23-Ubuntu SMP Mon Sep 24 16:31:06 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

root@ip-10-12-89-151:~# cat /etc/*release*
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.1 LTS"
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

**Comment by fwanssa** [comment]
We are seeing the same issue with the latest agent docker image.

**Comment by albanpeignier** [comment]
We're experiencing the same issue (datadog-agent 6.15.0-1). The restarted containers are listed into datadog-agent status but no message is retrieved. A agent restart fixes the problem.

**Comment by sonork** [comment]
Downgrading datadog-agent to 6.13.0-1 fixed the issue for me.
On every version above 6.13.0-1 datadog-agent logged the following message after a ```docker-compose restart```:
```CORE | WARN | (pkg/logs/input/docker/launcher.go:198 in startTailer) | Can't tail twice the same container: $container_id```
The restarted containers also did not show up in output of ```datadog-agent configcheck```

**Comment by fwanssa** [comment]
This bug is affecting us as well. We are using datadog agent version 7.16.1.

**Comment by sgnn7** [comment]
There has been a lot of work in the logs part of the agent in the recent releases that may have fixed this issue. Since I don't see clear replication steps, can anyone in this thread confirm that this problem still exists with agent `v7.26.0` and/or provide some steps to reproduce it locally?

**Comment by sonork** [comment]
datadog-agent: 7.26.0
docker-ce: 19.03.4
os: ubuntu 18.04

No issues.

**Comment by sgnn7** [comment]
@sonork Great to hear!

I'll close this issue as it seems likely that it's resolved by upgrading to the latest version (v7.26.0+) but please feel free to reopen this issue if anyone notices this to still be a problem.
